internal/service/subscription_line_item.go

```diff
@@ -0,0 +1,105 @@
+package service
+
+import (
+	"context"
+	"time"
+
+	"github.com/flexprice/flexprice/internal/api/dto"
+	ierr "github.com/flexprice/flexprice/internal/errors"
+	"github.com/flexprice/flexprice/internal/types"
+)
+
+// AddSubscriptionLineItem adds a new line item to an existing subscription
+func (s *subscriptionService) AddSubscriptionLineItem(ctx context.Context, subscriptionID string, req dto.CreateSubscriptionLineItemRequest) (*dto.SubscriptionLineItemResponse, error) {
+	if err := req.Validate(); err != nil {
+		return nil, err
+	}
+
+	// Get the subscription
+	sub, err := s.SubRepo.Get(ctx, subscriptionID)
+	if err != nil {
+		return nil, err
+	}
+
+	// Validate subscription status
+	if sub.SubscriptionStatus != types.SubscriptionStatusActive {
+		return nil, ierr.NewError("subscription is not active").
+			WithHint("Only active subscriptions can have line items added").
+			WithReportableDetails(map[string]interface{}{
+				"subscription_id": subscriptionID,
+				"status":          sub.SubscriptionStatus,
+			}).
+			Mark(ierr.ErrValidation)
+	}
+
+	// Initialize line item params
+	params := dto.LineItemParams{
+		Subscription: sub,
+	}
+
+	// Get entity details and price with expanded data
+	switch {
+	case req.PriceID != "":
+		// Get base price to determine entity type
+		price, err := s.PriceRepo.Get(ctx, req.PriceID)
+		if err != nil {
+			return nil, err
+		}
+
+		// Set the price in params
+		params.Price = &dto.PriceResponse{Price: price}
+
+		if price.EntityType == types.PRICE_ENTITY_TYPE_PLAN {
+			planService := NewPlanService(s.ServiceParams)
+			planResponse, err := planService.GetPlan(ctx, price.EntityID)
+			if err != nil {
+				return nil, err
+			}
+			params.Plan = planResponse
+			params.EntityType = types.SubscriptionLineItemEntityTypePlan
+		} else {
+			addonService := NewAddonService(s.ServiceParams)
+			addonResponse, err := addonService.GetAddon(ctx, price.EntityID)
+			if err != nil {
+				return nil, err
+			}
+			params.Addon = addonResponse
+			params.EntityType = types.SubscriptionLineItemEntityTypeAddon
+
+		}
+	}
```

Comment:
refactor clean - throw error on req.PriceID  = empty and then use switch cases for each entity type